### PR TITLE
feat(comments): add report comments functionality, restrict report twice

### PR DESCRIPTION
**Pull request by VERZUOL1**

### CODE REVIEW

File: astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/Comment.tsx [hunk 1]
```diff
@@ -112,6 +118,20 @@ export const Comment: FC<CommentProps> = ({
...
 
+  function renderReportedMessage() {
+    if (!reportsCount) {
+      return null;
+    }
+
+    return (
+      <div className={styles.reportedInfo}>
+        {isReported
+          ? `${t('comments.youReportedThisMessage')} ${reportsCount}/4`
```
Comment (AlexeyPonomarenko) [inline astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/Comment.tsx]: Is there sense to move `${reportsCount}/4` to variable and reuse it?

File: astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/Comment.tsx [hunk 2]
```diff
@@ -141,17 +165,19 @@ export const Comment: FC<CommentProps> = ({
...
+          <div className={styles.commentControlButtonWrapper}>
+            <ActionButton
+              className={styles.commentControlButton}
+              tooltipPlacement="top"
+              iconClassName={styles.commentControlIcon}
+              iconName="commentBlock"
+              onClick={handleReportAction}
+              tooltip="Report"
+            />
+          </div>
```
Comment (AlexeyPonomarenko) [inline astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/Comment.tsx]: Is there sense to move it to render method, like
```
function renderActionButton(icon: IconName, action: () => void, tooltip) {
  return (
    <div className={styles.commentControlButtonWrapper}>
            <ActionButton
              className={styles.commentControlButton}
              tooltipPlacement="top"
              iconClassName={styles.commentControlIcon}
              iconName={icon}
              onClick={action}
              tooltip={tooltip}
            />
          </div>
  )
}
```

and reuse it in line 158 and 169?

File: astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/Comment.tsx (hunk 1 above)
Comment (VERZUOL1) [reply astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/Comment.tsx]: good catch! fixed

File: astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/Comment.tsx (hunk 2 above)
Comment (VERZUOL1) [reply astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/Comment.tsx]: done!

**Review by AlexeyPonomarenko — APPROVED** [review]

**Pull request merged**